### DISCUSSION TITLE
Training results for Russian 22050Hz Natasha

### DISCUSSION THREAD

**Discussion by shigabeev** [p0p4k/vits2_pytorch#32] [Show and tell]
The model has trained for only 58,000 steps with no sdp and no duration discriminator.  The number of text encoder layers was increased from 6 to 10.

The model, a sound sample and symbols.py for russian can be found on Google drive.

https://drive.google.com/drive/folders/1v-jGF8k_gfIUHHFafA1qGOtzDm3YlVbs?usp=sharing

**Comment by beqabeqa473** [comment]
cool. It seems it needs more steps to fix sounding. looks like vocoder is not trained to that point

**Comment by shigabeev** [comment]
This time I trained it with SDP for 138K epochs. Now it sounds far better.

https://drive.google.com/drive/folders/1y8cDIp0MmSP2LS6V8jZ7fjKLKfw33kas?usp=sharing

**Reply by p0p4k** [reply]
Also with dp-discriminator?

**Reply by shigabeev** [reply]
It turns out that your setup doesn't turn off DP discriminator no matter what's written in config, so luckily it has always been turned on. It works!

**Reply by p0p4k** [reply]
Oops, I'll look into that today. Thanks for letting me know.

**Comment by beqabeqa473** [comment]
Also, is it still using hifigan vocoder or you used work from your branch?

On 8/30/23, p0p ***@***.***> wrote:
> Also with dp-discriminator?
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/p0p4k/vits2_pytorch/discussions/32#discussioncomment-6865498
> You are receiving this because you commented.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Reply by shigabeev** [reply]
I tried HifiGAN, MS-iSTFT and bigVGAN. Sound quality for single speaker would be ranked the following way:

1. MS-iSTFT (the best)
2. HFG
3. BigVGAN

I believe that BigVGAN addresses different problems, mainly generalization on multi-speaker + multi-language setup. So it takes much longer to train it, results sound worse and the inference time is longer.

**Reply by p0p4k** [reply]
MS-iSTFT (the best) <-- with vits2?

**Reply by shigabeev** [reply]
Yes, there is a fork of your repo with MS-iSTFT. It sounds majestic. At least, on my data.

https://github.com/FENRlR/MB-iSTFT-VITS2

**Reply by p0p4k** [reply]
Should we mix this repo with that, so users get to choose more vocoder options?

**Reply by p0p4k** [reply]
@shigabeev Also a few samples to compare all three would be great , if you can manage that! Thanks and I appreciate your efforts!!!

**Reply by shigabeev** [reply]
I halted the BigVGAN training at around 45K steps due to its performance being notably inferior to both HFG and iSTFT at that epoch. However, this comparison might not be fully accurate since the batch size varied; BigVGAN consumed 4x the memory compared to iSTFT, prompting an adjustment in the learning rate.

Despite this, I believe given another week, BigVGAN might yield comparable results. Nonetheless, I'm currently inclined towards iSTFT due to its satisfactory performance.

**Audio Comparisons:**

HFG (190K steps) vs. iSTFT (190K steps)
BigVGAN (40K steps) vs. iSTFT (40K steps) 

[Link to audios.](https://drive.google.com/drive/folders/1V22Kefp03rQrr5jBqDThCXnWZ0E5IHHT?usp=sharing)

**Reply by p0p4k** [reply]
Thanks for your reply. I will consider adding these vocoder options in this repo as well.

**Comment by Subarasheese** [comment]
@shigabeev Would you mind sharing what were your steps to organize the dataset and prepare it for training? I am trying to train a model on my native language but I am facing issues, as seen here:

https://github.com/p0p4k/vits2_pytorch/issues/33

**Reply by shigabeev** [reply]
I can record a stream on youtube if you'll watch.

**Reply by jswildone** [reply]
@shigabeev If you did that would be absolutely incredible. Happy to send some money to help you fund the training if so

**Reply by bzp83** [reply]
hi @shigabeev,
did you record a video? would you mind sharing? thank you!

**Comment by beqabeqa473** [comment]
It would be cool to see samples.

On 9/1/23, p0p ***@***.***> wrote:
> Oops, I'll look into that today. Thanks for letting me know.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/p0p4k/vits2_pytorch/discussions/32#discussioncomment-6884260
> You are receiving this because you commented.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by Xmiler** [comment]
@shigabeev hi!
Thanks for sharing your result. Could you tell us the way you pnonemized your dataset please :pray:

**Comment by shigabeev** [comment]
Natasha is already normalized and with accents. So I just added trained NN on it as it is, on graphemes.
________________________________
